File: MyHordesOptimizerWebsite/src/app/_abstract_model/dto/expedition.dto.ts
import { ExpeditionPartDTO } from './expedition-part.dto';

export interface ExpeditionDTO {
    id?: number;
    state: 'ready' | 'stop';
    label: string;
    minPdc: number;
    position: number;
    parts: ExpeditionPartDTO[];
}

export interface ExpeditionShortDTO {
    id?: number;
    state: 'ready' | 'stop';
    label: string;
    minPdc: number;
    position: number;
    partsId: number[];
}

File: MyHordesOptimizerApi/MyHordesOptimizerApi/Dtos/MyHordesOptimizer/Expeditions/ExpeditionDto.cs
using System.Collections.Generic;

namespace MyHordesOptimizerApi.Dtos.MyHordesOptimizer.Expeditions
{
    public class ExpeditionDto
    {
        public int? Id { get; set; }
        public string State { get; set; }
        public string? Label { get; set; }
        public int? MinPdc { get; set; }
        public List<ExpeditionPartDto>? Parts { get; set; }
    }
}
